src/accelerate/optimizer.py

```diff
@@ -121,6 +121,18 @@ def zero_grad(self, set_to_none=None):
                     raise ValueError("`set_to_none` for Optimizer.zero_grad` is not supported by this optimizer.")
                 self.optimizer.zero_grad()
 
+    def train(self):
+        """
+        Sets the optimizer to "train" mode. Useful for optimizers like `schedule_free`
+        """
+        return self.optimizer.train()
+
+    def eval(self):
+        """
+        Sets the optimizer to "eval" mode. Useful for optimizers like `schedule_free`
+        """
+        return self.optimizer.eval()
+
```

Comment:
How about adding a `@property` for `self.optimizer.training` too? I don't think we also need a setter for this, as `train()` and `eval()` should be enough.

Reply:
That gets to be optimizer-specific, so not a fan of it unless its downstreamed, as they currently don't have that: https://github.com/facebookresearch/schedule_free/blob/main/schedulefree/adamw_schedulefree.py#L86

(Otherwise I'd agree, yes that's a good idea)

Reply:
Ah I see, good point.